### PR TITLE
[JUJU-3489] Support requires-prompts mode model-config

### PR DESCRIPTION
**Pull request by jack-w-shaw**
This will enable forwards compatibility with future controllers, particularly useful for model migrations

This will require versions 3.2 to bump their minimum migration source versions to 2.9.43

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- [ ] Go unit tests, with comments saying what you're testing
- ~[ ] [Integration tests](https://github.com/juju/juju/tree/develop/tests), with comments saying what you're testing~
- ~[ ] [doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages~

## QA steps

```sh
$ juju-2.9 bootstrap lxd lxd-2.9
$ juju-2.9 add-model m
$ juju-2.9 deploy ubuntu -n2
$ juju-3.2 bootstrap lxd lxd-3.2
$ juju-2.9 switch lxd-2.9
$ juju-2.9 migrate m lxd-3.2
$ juju-3.2 switch lxd-3.2:admin/m
$ juju-3.2 add-unit ubuntu
```
(verify the extra unit is added successfully)

## Bug reference

https://bugs.launchpad.net/juju/+bug/2015398

### CODE REVIEW

File: environs/config/config.go [hunk 1]
```diff
@@ -1432,6 +1433,7 @@ func (c *Config) validateMode() error {
 	for _, mode := range modes {
 		switch strings.TrimSpace(mode) {
 		case "strict":
+		case "requires-prompts":
```
Comment (tlm) [inline environs/config/config.go:1436]: Can we have a test for this please ?

**Review by tlm — CHANGES_REQUESTED** [review]

File: environs/config/config.go (hunk 1 above)
Comment (jack-w-shaw) [reply environs/config/config.go:1436]: Yes of course

File: environs/config/config.go [hunk 2]
```diff
@@ -1404,8 +1404,9 @@ func (c *Config) validateCharmHubURL() error {
 }
 
 // Mode returns the mode type for the configuration.
-// Only two modes exist at the moment (strict or ""). Empty string
+// Only two modes are supported at the moment (strict or ""). Empty string
```
Comment (anvial) [inline environs/config/config.go:1407]: So, means "only three modes ..."

**Review by anvial — APPROVED** [review]
QA passed, 

LG2T, but I think we should update the comment (about the number of "modes").

File: environs/config/config.go (hunk 2 above)
Comment (jack-w-shaw) [reply environs/config/config.go:1407]: Good spot

**Review by tlm — APPROVED** [review]
lgtm

**Pull request merged**